spec/unit/MatrixBanSync.spec.js

```diff
@@ -0,0 +1,151 @@
+const { MatrixGlob } = require("matrix-bot-sdk");
+const { MatrixBanSync } = require("../../lib/bridge/MatrixBanSync");
+
+const BANNED_USER_STATE_EVENT = {
+    type: "m.policy.rule.user",
+    state_key: 'banned-user',
+    content: {
+        recommendation: 'm.ban',
+        entity: '@user:banned.com',
+        reason: 'badly-behaved',
+    }
+};
+
+const BANNED_USER_ENTITY = {
+    matcher: new MatrixGlob(BANNED_USER_STATE_EVENT.content.entity),
+    entityType: BANNED_USER_STATE_EVENT.type,
+    reason: BANNED_USER_STATE_EVENT.content.reason
+};
+
+const BANNED_SERVER_STATE_EVENT = {
+    type: "m.policy.rule.server",
+    state_key: 'banned-server',
+    content: {
+        recommendation: 'm.ban',
+        entity: 'banned-server.com',
+        reason: 'badly-constructed',
+    }
+};
+
+const BANNED_SERVER_ENTITY = {
+    matcher: new MatrixGlob(BANNED_SERVER_STATE_EVENT.content.entity),
+    entityType: BANNED_SERVER_STATE_EVENT.type,
+    reason: BANNED_SERVER_STATE_EVENT.content.reason
+};
+
+let banSync;
+
+describe("MatrixBanSync", () => {
+    beforeEach(() => {
+        banSync = new MatrixBanSync({ rooms: [] });
```

Comment:
I wonder if mocha tests can have local variables.
With `banSync` being reused and mutated the tests cannot run in parallel.
Then again, running tests in parallel is not a requirement we have to fulfill.

Reply:
(it's not even mocha, it's :sparkle: jasmine :sparkle:)

But yeah, one day a big refactor will be due.